### PR TITLE
node package #444

### PR DESCRIPTION
**Pull request by AlaaElattar**
### Issues

- https://github.com/threefoldtech/terraform-provider-grid/issues/444

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 1]
```diff
@@ -1,43 +1,72 @@
...
 ```hcl
 # Copy-paste your Terraform configurations here - for large Terraform configs,
 # please use a service like Dropbox and share a link to the ZIP file. For
 # security, you can also encrypt the files using our GPG public key.
 ```
 
 ### Debug Output
-Please provider a link to a GitHub Gist containing the complete debug output: https://www.terraform.io/docs/internals/debugging.html. Please do NOT paste the debug output in the issue; just paste a link to the Gist.
+
+Please provider a link to a GitHub Gist containing the complete debug output: <https://www.terraform.io/docs/internals/debugging.html>. Please do NOT paste the debug output in the issue; just paste a link to the Gist.
```
Comment (AbdelrahmanElawady) [inline .github/ISSUE_TEMPLATE/bug_report.md:37]: typo `provider` should be `provide`

File: internal/node/proxy.go [hunk 2]
```diff
@@ -35,8 +36,8 @@ type ProxyBus struct {
 }
 
 func NewProxyBus(endpoint string, twinID uint32, sub subi.SubstrateExt, signer substrate.Identity, verifyReply bool) (*ProxyBus, error) {
-	if len(endpoint) != 0 && endpoint[len(endpoint)-1] == '/' {
-		endpoint = endpoint[:len(endpoint)-1]
+	if len(endpoint) != 0 {
```
Comment (AbdelrahmanElawady) [inline internal/node/proxy.go:39]: i think the check for endpoint length here isn't needed `TrimSuffix` will return an empty string in that case

**Review by AbdelrahmanElawady — COMMENTED** [review]
looks great, just missing the validation part and comments on exported function from #444

File: internal/node/client_pool.go [hunk 3]
```diff
@@ -6,32 +6,37 @@ import (
...
+		clients: make(map[uint32]*NodeClient),
+		rmb:     rmb,
 	}
 }
 
-func (k *NodeClientPool) GetNodeClient(sub subi.SubstrateExt, nodeID uint32) (*NodeClient, error) {
-	cl, ok := k.nodeClients[nodeID]
+// GetNodeClient gets the node client according to node ID
+func (p *NodeClientPool) GetNodeClient(sub subi.SubstrateExt, nodeID uint32) (*NodeClient, error) {
+	cl, ok := p.clients[nodeID]
```
Comment (xmonader) [inline internal/node/client_pool.go:30]: lines can be merged together if cl, ok:= ; ok { return cl, nil}

**Review by xmonader — CHANGES_REQUESTED** [review]
Looks good, let's do another minor fix twinId -> twinID and should be good to merge

File: internal/node/client_pool.go (hunk 3 above)
Comment (rawdaGastan) [reply internal/node/client_pool.go:30]: cl is used below

**Review by xmonader — APPROVED** [review]

**Pull request merged**